src/mbgl/util/grid_index.cpp

```diff
@@ -263,12 +263,12 @@ void GridIndex<T>::query(const BCircle& queryBCircle, std::function<bool (const
 
 template <class T>
 std::size_t GridIndex<T>::convertToXCellCoord(const float x) const {
-    return util::max(size_t(0), util::min(xCellCount - 1, static_cast<size_t>(std::floor(x * xScale))));
```

Comment:
This one happens if `x` is negative. Thinking of a most elegant fix.

I also found another set of warnings that we can enable which will help us detect signed vs unsigned issues. Will investigate how many warnings there are.